MoltenVK/MoltenVK/GPUObjects/MVKImage.mm

```diff
@@ -561,6 +561,161 @@
     }
 }
 
+template<typename ImgRgn>
+static MTLRegion getMTLRegion(const ImgRgn& imgRgn) {
+	return { mvkMTLOriginFromVkOffset3D(imgRgn.imageOffset), mvkMTLSizeFromVkExtent3D(imgRgn.imageExtent) };
+}
+
+// Host-copy from memory to a MTLTexture.
+VkResult MVKImage::copyContent(id<MTLTexture> mtlTex,
+							   VkImageToMemoryCopyEXT imgRgn, uint32_t mipLevel, uint32_t slice,
+							   void* pImgBytes, size_t rowPitch, size_t depthPitch) {
+	[mtlTex getBytes: pImgBytes
+		 bytesPerRow: rowPitch
+	   bytesPerImage: depthPitch
+		  fromRegion: getMTLRegion(imgRgn)
+		 mipmapLevel: mipLevel
+			   slice: slice];
+	return VK_SUCCESS;
+}
+
+// Host-copy from a MTLTexture to memory.
+VkResult MVKImage::copyContent(id<MTLTexture> mtlTex,
+							   VkMemoryToImageCopyEXT imgRgn, uint32_t mipLevel, uint32_t slice,
+							   void* pImgBytes, size_t rowPitch, size_t depthPitch) {
+	VkSubresourceLayout imgLayout = { 0, 0, rowPitch, 0, depthPitch};
```

Comment:
If I am not mistaken the two comments here are swapped.

Reply:
Thanks. Fixed now.